#2547 - fix image on CMS page styled different from Magento admin styling

tatiana-scandi opened the pull request:
Issue: https://github.com/scandipwa/scandipwa/issues/2547

It was very hard to select approach to fix this issue, as nobody in the team was able to explain why we are reformatting images in HTML layout.

E.g. what should be (the version that corresponds to what I set in Magento admin):
https://monosnap.com/file/ASF2DZbOMQzi4fdmhgYu96m9BaClzF

What we get after img is transformed to <Image/> and lots of additional styles are added:
https://monosnap.com/file/sBkVD470ml188UqcrEL43cXGQl9IMm

First, image is not centered.
Second, we are trying to fit the image into square (this is default `<Image/>` logic).

And yes, current `<Image/>` Component works perfect when is comes to fitting image into a limited sized container. But it does harm, when freely positioned image in an article.  

So, I had 3 options to approach this issue:
1. Rewrite Image component logic of adding styles based on passed props, so that we get effect like there were no additional styles. 2 problems here:
- why overcomplicate and rewrite logic to remove styles, if can just not add them. I was trying to add such logic, but it really looked like removal of all `<Image/>` styles to achieve the desired effect.
- risk to affect images for which `<Image/>` works fine with current implementation

2. Just not use `<Image/>` at all. Here is an issue that we receive `height` and `width` as separate props in html layout, not inside style object. Also, if `width` and `height` are in `px`, you receive just numeric value, and have to add `px` yourself. And check that it is numeric, not value in percent or smth like `auto`. I tried to write such a logic, but it looked quite ugly:
```
    replaceImages({ attribs }) {
        const attributes = attributesToProps(attribs);

        if (!attribs.src) {
            return null;
        }

        const {
            alt = '',
            width: initialWidth = '',
            style,
            ...otherAttributes
        } = attributes;

        const width = isNaN(initialWidth) ? initialWidth : ${initialWidth}px;

        if (attribs.src) {
            return <img alt={ alt } style={ { width, ...style } } { ...otherAttributes } />;
        }
    }
```
 Moreover, such logic already exists in `<Image/>` + covers more cases, so why rewrite it? Plus, `<Image/>` has error handling, managing image loading states, placeholders, which are useful for any image which you would like to render.

So what I finally came up with:
1. Still use `<Image/>` for html content.
2. Use additional render logic provided by `<Image/>`
3. Don't add image container and additional styles to keep image formatting as is.

What I also did, is I got rid of `switch` in `<Image/>` component, which makes it hard to override logic.
I had to replace constant values in config to string ones, as they are used as object keys, and if numbers are used, we need additional type conversion (numbers as object keys are autoconverted to strings).

Another PR with attempt to resolve same issue: https://github.com/scandipwa/scandipwa/pull/2560

aleksandrakorolova commented:
1. Images are missing in category preview block
![Screenshot 2021-08-09 at 10 26 08](https://user-images.githubusercontent.com/73526235/128672223-344a1333-6940-4d3d-bb15-f3270eb18ed8.png)
2. Mobile: Page has horizontal scroll
<img src="https://user-images.githubusercontent.com/73526235/128672413-8c4f7719-3281-4f88-8a0c-711a75e0d12e.jpg" width="400" height="1000">
<img src="https://user-images.githubusercontent.com/73526235/128672417-5a1c2ea5-3371-4c57-ae15-4381ee7d44e2.jpg" width="400" height="1000">
3. Opera: Title is shifted
<img src="https://user-images.githubusercontent.com/73526235/128672453-4334eaec-f710-4949-9b4b-2b2faa5d4e7f.jpg" width="400" height="1000">

tatiana-scandi commented:
Issue 1: fixed issue with classname lost

Issue 2:
html tag for image initially looked like:
```<img style="display: block; margin-left: auto; margin-right: auto;" src="{{media url="wysiwyg/maxresdefault.jpg"}}" alt="" width="700" height="500">```

So, we want width of 700px and it is correctly being applied. We see horizontal scroll when viewport width is less than 700px. I changed width of 700px to max-width of 700px => it makes image width not exceed viewport width.

Issue 3:
I was not able to reproduce issue on Android, Opera browser:
![photo_2021-08-09_11-31-48](https://user-images.githubusercontent.com/79456428/128693880-640f4954-4368-4e57-9a50-ac0d7f6b011d.jpg)
Please, recheck (probably was fixed by issue 2 changes)
